pandas/io/json/_json.py

```diff
@@ -352,12 +352,12 @@ def read_json(
 
     Parameters
     ----------
-    path_or_buf : a valid JSON string or file-like, default: None
+    path_or_buf : a valid JSON string or file-like, default None
         The string could be a URL. Valid URL schemes include http, ftp, s3,
         gcs, and file. For file URLs, a host is expected. For instance, a local
-        file could be ``file://localhost/path/to/table.json``
+        file could be ``file://localhost/path/to/table.json``.
 
-    orient : string,
+    orient : str, default None
```

Comment:
According to the description of the parameter, the default is columns. I guess `None` means columns here. I think the signature should probably be changed if I'm not missing something, but I'd rather have an `optional` or `default 'columns'` than `default None`.

Do you mind reviewing the other `default None` in the PR too, and see whether they should be `optional` instead?

Reply:
Assuming the default value is used, it looks like `"columns"` will be bound to `orient` when `_get_object_parser` is called in `JsonReader.read()`. `_get_object_parser` creates a `FrameParser` object, which results in the `Parser` init being called, where the `"columns"` assignment happens. But before all this happens, `orient` is used while it holds `None`. I haven't looked at the code enough to see whether changing the signature would have any unintended consequences.

I will make the change to `optional`. Just to explain my reasoning for choosing `default None`, I figured it would be appropriate because it's merely acting as a fill value. It doesn't mean no orientation, like `chunksize=None` means do not chunk.

I did review the other `default None` uses before my last commit. I have reviewed them again just to see if I missed something. I'm not sure why path_or_buf has a default value of `None` other than to make it non-positional. It's a necessary argument or the function blows up. For the rest of the `default None` uses, it appears to be a fill value unless my understanding is wrong.